File: 프로그래머스/2/12953. N개의 최소공배수/N개의 최소공배수.js
function solution(arr) {
    const GCD = (a, b) => {
        while(b) {
            [a, b] = [b, a % b];    
        }
        return a;
    }
    
    const LCM = (a, b) => {
        return Math.floor((a * b) / GCD(a, b));
    }
    
    let result = arr[0];
    for(let i = 1 ; i < arr.length ; i++){
        result = LCM(result, arr[i]);
    }
    
    return result;
}

File: 프로그래머스/2/12953. N개의 최소공배수/N개의 최소공배수.py
def GCD(a, b):
    while b:
        a, b = b, a % b
    return a

def LCM(a, b):
    result = (a * b) // GCD(a, b)
    return result

def solution(arr):
    result = arr[0]
    for i in range(1, len(arr)):
        result = LCM(result, arr[i])
    return result
    